app/server/app/routes/api.js

```diff
@@ -44,17 +44,27 @@ async function queryColumnValues(profile, column, params, schema) {
   Object.entries(params).forEach(([name, value]) => {
     if (name === "text") parsedParams.text = value;
     else if (name === "limit") parsedParams.limit = value;
+    else if (name === "direction") parsedParams.direction = value;
     else parsedParams.filters[name] = value;
   });
 
   const query = knex
     .withSchema(schema)
     .from(profile.tableName)
     .column(column.name)
-    .whereILike(column.name, `%${parsedParams.text}%`)
     .distinctOn(column.name)
+    .orderBy(column.name, parsedParams.direction ?? "asc")
     .select();
 
+  if (column.type === "numeric" || column.type === "timestamptz") {
+    query.whereRaw("CAST(?? as TEXT) ILIKE ?", [
+      column.name,
+      `%${parsedParams.text}%`,
+    ]);
+  } else {
+    query.whereILike(column.name, `%{parsedParams.text}%`);
+  }
```

Comment:
```suggestion
  if (parsedParams.text) {
    if (column.type === "numeric" || column.type === "timestamptz") {
      query.whereRaw("CAST(?? as TEXT) ILIKE ?", [
        column.name,
        `%${parsedParams.text}%`,
      ]);
    } else {
      query.whereILike(column.name, `%{parsedParams.text}%`);
    }
  }
```

We can skip the `whereILike` if there is no text being provided. This should improve performance a little bit.

Reply:
Good call!